res/controllers/Pioneer-DDJ-200-scripts.js

```diff
@@ -19,11 +19,36 @@ DDJ200.init = function() {
 
         var vgroup = "[Channel" + i + "]";
 
-        // run onTrackLoad after every track load to set LEDs accordingly
+        // run updateDeckLeds after every track load to set LEDs accordingly
         engine.makeConnection(vgroup, "track_loaded", function(ch, vgroup) {
-            DDJ200.onTrackLoad(ch, vgroup);
+            DDJ200.updateDeckLeds(vgroup);
         });
 
+        // run switchPlayLED after play/pause track to set LEDs accordingly
+        engine.makeConnection(vgroup, "play", function(ch, vgroup) {
+            var vDeckNo = script.deckFromGroup(vgroup);
+            var d = (vDeckNo % 2) ? 0 : 1;
+            DDJ200.switchPlayLED(d, ch);
+        });
+
+        // run switchSyncLED after sync toogle to set LEDs accordingly
+        engine.makeConnection(vgroup, "sync_enabled", function(ch, vgroup) {
+            var vDeckNo = script.deckFromGroup(vgroup);
+            var d = (vDeckNo % 2) ? 0 : 1;
+            DDJ200.switchSyncLED(d, ch);
+        });
+
+        // listen to changes on hotcues
+        for (var j = 1; j <= 8; j++) {
+            // run switchPadLED after every hotcue update
+            engine.makeConnection(vgroup, "hotcue_" + j + "_enabled", function(ch, vgroup, control) {
+                var pad = Number(control.split("_")[1]);
+                var vDeckNo = script.deckFromGroup(vgroup);
+                var d = (vDeckNo % 2) ? 0 : 1;           // d = deckNo - 1
+                DDJ200.switchPadLED(d, pad, ch);
```

Comment:
Let's not use one-character variable names (except `i`/`j` for counters maybe) and go with something more meaningful:

```suggestion
                var deckNo = script.deckFromGroup(vgroup);
                var deckOffset = (deckNo - 1) % 2;
                DDJ200.switchPadLED(deckOffset, pad, ch);
```

I know the rest of the script does that too, but let's at least fix it for new code :D

Reply:
I don't like one-character variables too, but at same time is good to keep a pattern at the code. Don't you think would be better open a new PR fixing all one-character variables after this one?

Reply:
I second @Holzhaus, and I think it should probably be done in this PR

Reply:
Just the new code or a full refactor?

Reply:
`var d = (vDeckNo % 2) ? 0 : 1;           // d = deckNo - 1`
seems clear and short to me.

`var deckOffset = (deckNo - 1) % 2;`
is neither short nor clear.
It is less obvious that the value is either 0 or 1.
And it is not correct, because it is not an offset. I don't know what to call this, but for sure not offset.

Reply:
`var controllerDeckIndex = (vDeckNo % 2) ? 0 : 1;`?
Case this represents the controller deck and an index starting from 0 to 1